src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/validation.ts

```diff
@@ -942,8 +958,23 @@ export function mergeValidationObjects(...sources: IValidations[]): IValidations
           if (!validations[layout][component][binding]) {
             validations[layout][component][binding] = { errors: [], warnings: [] };
           }
-          validations[layout][component][binding].errors = validations[layout][component][binding].errors.concat(source[layout][component][binding]?.errors);
-          validations[layout][component][binding].warnings = validations[layout][component][binding].warnings.concat(source[layout][component][binding]?.warnings);
+
```

Comment:
For readability, would it be possible to break these long selectors down to variables that describe what's going on?
Also, could we somehow extract some of this stuff into functions and add tests?

Reply:
@mjulstein I have re-written this now and broken it up into several functions to try and make it clear what exactly is happening here.